### ISSUE TITLE
Lost all my account names by upgrading to 8.0.0

### ISSUE THREAD

**Issue by leafcutterant** [MetaMask/metamask-extension#8887]
**Describe the bug**
Title.

**To Reproduce (REQUIRED)**

1. Have a lot of accounts (mine were all separately imported via private keys)
2. Give them names
3. Upgrade from something like 7.7.9 to 8.0.0

**Expected behavior**
My account names are left alone.

**Screenshots**
![fuck](https://user-images.githubusercontent.com/7106231/86392064-1ebee080-bc9b-11ea-9926-422f54bacb9b.png)

**Browser details (please complete the following information):**
 - Firefox 68.0
 - MetaMask 8.0.0

**Additional context (Error Messages, etc.)**
Until you do something like #2340 (2017) or #5913 (2018), you'll keep playing Russian roulette with user-generated data. I didn't even get an answer in either thread.

**Comment by tmashuang** [comment]
Is there any commonality about those addresses to determine the cause?

**Comment by leafcutterant** [comment]
Apart from the fact that they are accounts that were imported by pasting private keys, no commonality.

I don't think any accounts disappeared, but it's hard to tell since all my labels are gone.

**Comment by rekmarks** [comment]
@leafcutterant I am sorry that we never got back to you regarding your previous inquiries.

I posted a comment in #5913 with instructions on how to download your state logs: https://github.com/MetaMask/metamask-extension/issues/5913#issuecomment-653242462

It would be helpful for our troubleshooting if you could privately send me your state logs. I have the same handle on Twitter, Telegram, Discord, and Keybase as I do on GitHub. You can also join the ConsenSys Discord here: https://discord.gg/EC77pH

**Comment by rekmarks** [comment]
@leafcutterant  we're unfortunately not confident that we'll be able to ship a fix that restores the labels in a timely fashion. That said, the labels should still be contained your local MetaMask extension state, and you have the option to restore them manually.

To find the labels in state, please do the following:

1. Follow the instructions in this support article to download the MetaMask state as a `.json` file: https://metamask.zendesk.com/hc/en-us/articles/360015289832-How-to-Download-State-Logs-

2. Open the downloaded file, and find the key `lostIdentities`. The full JSON path is: `metamask.lostIdentities`

3. You should see an object with Ethereum address keys and object values of the form `{ address, name }`. All accounts and the missing names should be included in that object, and you can manually rename the accounts in the MetaMask UI.

I regret that we don't have a better solution, but this should provide you with a solution you can pursue immediately.

If you encounter any further issues, please do create that support issue. We're also leaving this issue open for the time being, and please feel free to ask any further questions here as well.

**Comment by leafcutterant** [comment]
@rekmarks thank you, I appreciate that you reached out and tried to help.

> Meanwhile, it would be helpful for our troubleshooting if you could send us your state logs via Zendesk

I ended up not submitting my state logs because sanitizing them looks like an insurmountable challenge. I'd appreciate if there would be an offline a tool that removes all personal information from state logs. Or at least a guide on which parts of the state logs one should copy to exclude personal information.

> the labels should still be contained your local MetaMask extension state, and you have the option to restore them manually.

Back then at 8.0.1 I exported my state logs, but it didn't contain my lost labels. Under `lostIdentities` I only saw the same replacement labels (Account 1, Account 2, etc). I eventually gave up, reinstalled Metamask and started building my account from scratch. So I can't confirm nor deny that this issue is fixed.

I remain confident that the only way to prevent future mess like this is to provide a way to make backups.

**Comment by Gudahtt** [comment]
I'll close this issue for now, as anyone who was unfortunately caught by this bug has probably seen this or given up on restoring their old account names by name. We never did come up with a way to automatically restore the names for affected users without also risking the restoration of intentionally deleted or obsolete identities from older versions. I'm very sorry for any inconvenience this has caused - I can imagine this must have been very frustrating.

>I remain confident that the only way to prevent future mess like this is to provide a way to make backups.

Thanks for raising this again; I agree that a backup would be a great way to safeguard against this happening again in the future. I've created a new issue to keep track of this: #10187

**Comment by Beasta** [comment]
> @leafcutterant while we are fixing the _root cause_ in 8.0.1 of the extension, we're unfortunately not confident that we'll be able to ship a fix that _restores_ the missing labels in a timely fashion. That said, the labels should still be contained your local MetaMask extension state, and you have the option to restore them manually.
> 
> To manually restore the labels, please do the following:
> 
> 1. Follow the instructions in this support article to download the MetaMask state as a `.json` file: https://metamask.zendesk.com/hc/en-us/articles/360015289832-How-to-Download-State-Logs-
> 2. Open the downloaded file, and find the key `lostIdentities`. The full JSON path is: `metamask.lostIdentities`
> 3. You should see an object with Ethereum address keys and object values of the form `{ address, name }`. All accounts and the missing names should be included in that object, and you can manually rename the accounts in the MetaMask UI.
> 
> I regret that we don't have a better solution, but this should provide you with a solution you can pursue immediately. For reference, `lostIdentities` may include accounts that you no longer have stored in MetaMask. If the number is intractable, you can find the accounts that are still stored in MetaMask (with the incorrect names) under `metamask.identities`.
> 
> If you encounter any further issues, please do create a support issue. We're also leaving this issue open until we determine if we can ship a fix. Please feel free to ask any further questions here as well.

that zendesk link is outdate - use https://metamask.zendesk.com/hc/en-us/articles/360015290092-How-to-download-state-logs

**Comment by AaronD1** [comment]
Hi. I have a bit of an obscure issue. I created 1000 accounts, and to make managing these accounts easier I wanted to name them from 1 - 1000. Problem is, I messed up at 198 - there are two 198s, and the result is that the account labels above that mistake are offset by 1. Manually renaming all of those accounts to fix the issue would be very tedious, so I thought that I could download the state logs and write a script to rename them. Turns out though that's there's no way for me to restore the account labels from the state logs - I actually thought that was the point of the state logs. 

Is there any solution to this that doesn't involve manually renaming them?
